planner/planner_APIs.go

```diff
@@ -440,6 +470,67 @@ func (planner *MyPlanner) getPlanningApi(ctx *gin.Context) {
 	utils.LogErrorWithLevel(planner.ResultHTMLTemplate.Execute(ctx.Writer, planningResp), utils.LogError)
 }
 
+func (planner *MyPlanner) getPlanDetails(c *gin.Context) {
+
+	var id string = c.Param("id")
+	iowrappers.Logger.Debugf("GET Route /plans/%s", id)
+
+	var planRecordRedisKey = strings.Join([]string{iowrappers.TravelPlanRedisCacheKeyPrefix, id}, ":")
+	cachePlanSolution, cacheErr := planner.RedisClient.GetCachePlan(c, planRecordRedisKey)
+	if cacheErr != nil {
+		iowrappers.Logger.Debugf("Key: %s does not exist!\n", planRecordRedisKey)
+		c.String(http.StatusBadRequest, cacheErr.Error())
+		return
+	}
+
+	// FIXME: [Rui] check why placeIDs are not stored in order in cached slot solution
+	const fixedPlaceKeyPrefix = "place_details:place_ID:"
+	var placeKey string
+	var destination string = "Dream Place"
+	if cachePlanSolution.Destination != (POI.Location{}) {
+		destination = strings.Title(cachePlanSolution.Destination.City) + ", " + strings.ToUpper(cachePlanSolution.Destination.Country)
+	}
+	var tripResp = TripDetailResp{
+		PlaceDetails:      make([]PlaceDetailsResp, 0),
+		ShownActive:       make([]bool, 0),
+		TravelDestination: destination,
+	}
+	for idx, placeId := range cachePlanSolution.PlaceIDs {
+		if idx >= 3 {
+			iowrappers.Logger.Panicf("Panic: too many requests (id=%d) to Place API!\n", idx)
+		}
+
+		placeKey = fixedPlaceKeyPrefix + placeId
+		cachePlaceDetails, cacheErr := planner.RedisClient.GetCachePlace(c, placeKey)
+		if cacheErr != nil {
+			c.String(http.StatusBadRequest, cacheErr.Error())
+			return
+		}
+
+		placeDetails := planner.getTripFromPlace(cachePlaceDetails)
+		tripResp.PlaceDetails = append(tripResp.PlaceDetails, placeDetails)
+		var isActive = false
+		if idx == 0 {
+			isActive = true
+		}
+		tripResp.ShownActive = append(tripResp.ShownActive, isActive)
+
+	}
+	iowrappers.Logger.Debugf("Trip Details:\n%v\n", tripResp.PlaceDetails)
+
+	// send data
+	utils.LogErrorWithLevel(planner.TripHTMLTemplate.Execute(c.Writer, tripResp), utils.LogError)
+}
+
+func (planner *MyPlanner) getTripFromPlace(place POI.Place) PlaceDetailsResp {
```

Comment:
This method can be put in a `transformers.go` file in the package.

Reply:
This function `getTripFromPlace` also requires access of *MyPlanner.PhotoClient.GetPhotoURL(), and it seems easiser to put `getTripFromPlace` in this file.